custom/enikshay/two_b_release_1/management/commands/enikshay_2b_case_properties.py

```diff
@@ -0,0 +1,496 @@
+            }.get(test.get_case_property('purpose_of_testing'), "")
+            props['rft_dstb_diagnosis'] = test.get_case_property('diagnostic_test_reason')
+            props['rft_dstb_followup'] = test.get_case_property('follow_up_test_reason')
+
+        return CaseStructure(
+            case_id=test.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "update": props,
+            },
+        )
+
+    def migrate_referral(self, referral, occurrences):
+        self.total_referrals += 1
+        prop = referral.get_case_property
+        props = {
+            'referral_initiated_date': (prop('referral_date') or prop('date_of_referral')),
+            'referred_to_name': prop('referred_to_location_name'),
+            'referred_by_name': prop('referred_by'),
+            'referral_rejection_reason_other_detail': prop('reason_for_refusal_other_detail'),
+            'referral_rejection_reason': prop('reason_for_refusal'),
+            'referral_closed_date': prop('acceptance_refusal_date'),
+            'accepted_by_name': prop('phi'),
+        }
+
+        if occurrences:
+            occurrence = max((case.opened_on, case) for case in occurrences)[1]
+            index_kwargs = {'indices': [CaseIndex(
+                occurrence,
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=CASE_TYPE_OCCURRENCE,
+            )]}
+        else:
+            index_kwargs = {}
+
+        return CaseStructure(
+            case_id=referral.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "update": props,
+            },
+            **index_kwargs
+        )
+
+    def migrate_trail(self, trail, occurrences):
+        self.total_trails += 1
+        if occurrences:
+            occurrence = max((case.opened_on, case) for case in occurrences)[1]
+            index_kwargs = {'indices': [CaseIndex(
+                occurrence,
+                identifier='parent',
+                relationship=CASE_INDEX_CHILD,
+                related_type=CASE_TYPE_OCCURRENCE,
+            )]}
+        else:
+            index_kwargs = {}
+
+        return CaseStructure(
+            case_id=trail.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "update": {},
+            },
+            **index_kwargs
+        )
+
+    def open_secondary_owners(self, drtb_hiv, person, occurrences):
+        self.total_secondary_owners += 1
+        if occurrences:
+            occurrence = max((case.opened_on, case) for case in occurrences)[1]
+            index_kwargs = {'indices': [CaseIndex(
+                occurrence,
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=CASE_TYPE_OCCURRENCE,
+            )]}
+        else:
```

Comment:
if this has no index, won't this case just be a stub with no information attached to it? Might be worth not creating it since it won't do anything, and I assume also won't show up anywhere.